artview/plugins/manual_unfold.py

```diff
@@ -0,0 +1,176 @@
+        self.positiveButton.clicked.connect(self.positiveUnfold)
+        self.layout.addWidget(self.positiveButton, 3, 1)
+
+        self.negativeButton = QtGui.QPushButton("Unfold Negative Values")
+        self.negativeButton.clicked.connect(self.negativeUnfold)
+        self.layout.addWidget(self.negativeButton, 4, 1)
+
+        self.show()
+
+    def getFieldNames(self):
+        vel = str(self.velField.text())
+        if vel == '':
+            vel = pyart.config.get_field_name('velocity')
+        if vel not in self.Vradar.value.fields:
+            common.ShowWarning(
+                "vel_field not in Radar, can not perform correction")
+
+        corrVel = str(self.corrVelField.text())
+        if corrVel == '':
+            corrVel = pyart.config.get_field_name('corrected_velocity')
+
+        return vel, corrVel
+
+    def positiveUnfold(self):
+        self.unfold('positive')
+
+    def negativeUnfold(self):
+        self.unfold('negative')
+
+    def unfold(self, side):
+        # test radar
+        radar = self.Vradar.value
+        points = self.Vpoints.value
+        if radar is None:
+            common.ShowWarning("Radar is None, can not perform correction")
+            return
+
+        if points is None:
+            common.ShowWarning("Points is None, can not perform correction")
+            return
+
+        if ('ray_index' not in points.axes or
+            'range_index' not in points.axes):
+            common.ShowWarning(
+                "Error: Points must have axes 'ray_index' and 'range_index'")
+            return
+
+        vel, corrVel = self.getFieldNames()
+        data = radar.fields[vel]['data'].copy()
+        ray = points.axes['ray_index']['data']
+        rng = points.axes['range_index']['data']
+        nyquist = self.nyquistVelocity.value()
+        if side == 'positive':
+            data[ray,rng] = np.where(data[ray,rng]>0,
+                                     -2*nyquist+data[ray,rng], data[ray,rng])
+        elif side == 'negative':
+            data[ray,rng] = np.where(data[ray,rng]<0,
+                                     2*nyquist+data[ray,rng], data[ray,rng])
+
+        strong_update = False  # insertion is weak, overwrite strong
+        if corrVel in self.Vradar.value.fields.keys():
+            resp = common.ShowQuestion(
+                "Field %s already exists! Do you want to over write it?" %
+                corrVel)
+            if resp != QtGui.QMessageBox.Ok:
+                return
+            else:
+                strong_update = True
+
+        radar.add_field_like(vel, corrVel, data, replace_existing=True)
+        self.Vradar.update(strong_update)
+
+    def newRadar(self, variable, value, strong):
+        '''respond to change in radar.'''
+        if self.Vradar.value is None:
+            return
+
+        nyquist_vel = self.Vradar.value.get_nyquist_vel(0, True)
+        self.nyquistVelocit.setValue(nyquist_vel)
```

Comment:
you are right, but this just break the automatic recognition of nyquist_vel, the plugin should still be working.